### PR TITLE
Make figure in AR2 example reproducible

### PR DESCRIPTION
**Pull request by damonbayer**
This PR adds back in code to produce the ar2 plot in the ar2 example. This is based on the code that was removed in https://github.com/pyro-ppl/numpyro/pull/1350. It was last available in https://github.com/pyro-ppl/numpyro/blob/c5b1a9229daf964a98eb12a8ad905e0d0a6c9437/examples/ar2.py.

The figure was generated with:
```bash
python ar2.py --num-samples 1000 --num-warmup 1000 --num-chains 4
```

Closes https://github.com/pyro-ppl/numpyro/issues/1815.

### CODE REVIEW

**Review by fehiepsi — APPROVED** [review]
Thanks, @damonbayer!

**Comment by damonbayer** [comment]
Investigating test failures.

**Comment by damonbayer** [comment]
CI checks should now pass. Sorry about that.

File: examples/ar2.py [hunk 1]
```diff
@@ -64,8 +64,6 @@ def transition(carry, _):
     with numpyro.handlers.condition(data={"y": y[2:]}):
         _, mu = scan(transition, init, timesteps)
 
-    numpyro.deterministic("mu", mu)
```
Comment (fehiepsi) [inline examples/ar2.py]: Is this intended?

Comment (damonbayer) [reply examples/ar2.py]: No 🤦‍♂️

Comment (damonbayer) [reply examples/ar2.py]: Fixed in https://github.com/pyro-ppl/numpyro/pull/1816/commits/8ebbaa7bf4a072c2fd4e60ef0eaef3abfad6795e

**Comment by damonbayer** [comment]
@fehiepsi I don't think any of the test failures are related to my contribution. Please let me know if you disagree.

**Comment by fehiepsi** [comment]
Yes they are unrelated - @juanitorduz is working on them in #1817

**Comment by fehiepsi** [comment]
@damonbayer Could you sync with master? They are fixed upstream.

**Comment by damonbayer** [comment]
@fehiepsi Fork is synced!

**Pull request merged**